core/services/keystore/eth.go

```diff
@@ -225,33 +227,57 @@ func (ks *eth) IncrementNextSequence(address common.Address, chainID *big.Int, c
 	return nil
 }
 
-func (ks *eth) Enable(address common.Address, chainID *big.Int, qopts ...pg.QOpt) error {
+func (ks *eth) Add(address common.Address, chainID *big.Int, qopts ...pg.QOpt) error {
 	ks.lock.Lock()
 	defer ks.lock.Unlock()
 	_, found := ks.keyRing.Eth[address.Hex()]
 	if !found {
-		return errors.Errorf("no key exists with ID %s", address.Hex())
+		return ErrKeyNotFound
 	}
-	return ks.enable(address, chainID, qopts...)
+	return ks.addKey(address, chainID, qopts...)
 }
 
 // caller must hold lock!
-func (ks *eth) enable(address common.Address, chainID *big.Int, qopts ...pg.QOpt) error {
+func (ks *eth) addKey(address common.Address, chainID *big.Int, qopts ...pg.QOpt) error {
 	state := new(ethkey.State)
 	sql := `INSERT INTO evm_key_states (address, next_nonce, disabled, evm_chain_id, created_at, updated_at)
-VALUES ($1, 0, false, $2, NOW(), NOW()) ON CONFLICT (evm_chain_id, address) DO UPDATE SET
-disabled=false,
-updated_at=NOW()
-RETURNING id, next_nonce, address, evm_chain_id, disabled, created_at, updated_at;`
+			VALUES ($1, 0, false, $2, NOW(), NOW()) 
+			RETURNING *;`
 	q := ks.orm.q.WithOpts(qopts...)
 	if err := q.Get(state, sql, address, chainID.String()); err != nil {
 		return errors.Wrap(err, "failed to insert evm_key_state")
 	}
+	// consider: do we really need a cache of the keystates?
```

Comment:
@jmank88 what do you think?

Reply:
Possibly important for the current architecture, but @prashantkumar1982 wants to change this.